os/fs/smartfs/smartfs_smart.c

```diff
@@ -645,7 +662,16 @@ static int smartfs_sync_internal(struct smartfs_mountpt_s *fs, struct smartfs_of
 		}
 
 		sf->byteswritten = 0;
-		sf->bflags = 0;
+		/* File's data sector has been synced with MTD, now check if this is a new file entry and write the entry */
+		if (sf->bflags & SMARTFS_BFLAG_NEW_ENTRY) {
+			/* Flags for this entry have already been set and stored, so mode will not be used */
+			ret = smartfs_writeentry(fs, sf->entry, sf->entry.name, SMARTFS_DIRENT_TYPE_FILE, 0xFFFF, &sf->entry);
```

Comment:
Actually sf->entry.flags keeps mode too.
And smartfs_writeentry will reset flag & set mode again with this 0xFFFF...

Reply:
Ohkkay yes I missed this.
The check to see if flags are already valid was removed in your mail.
So some modifications required. Ill do the same.